internal/xds/matcher/matcher_header_test.go

```diff
@@ -467,3 +467,53 @@ func TestHeaderSuffixMatcherMatch(t *testing.T) {
 		})
 	}
 }
+
+func TestHeaderStringMatch(t *testing.T) {
+	tests := []struct {
+		name   string
+		key    string
+		sm     StringMatcher
+		md     metadata.MD
+		invert bool
+		want   bool
+	}{
+		{
+			name: "should-match",
+			key:  "th",
+			sm: StringMatcher{
+				exactMatch: newStringP("tv"),
+			},
+			md:     metadata.Pairs("th", "tv"),
+			invert: false,
+			want:   true,
+		},
+		{
+			name: "not match",
+			key:  "th",
+			sm: StringMatcher{
+				containsMatch: newStringP("tv"),
+			},
+			md:     metadata.Pairs("th", "not-match"),
+			invert: false,
+			want:   false,
+		},
+		{
+			name: "invert string match",
+			key:  "th",
+			sm: StringMatcher{
+				containsMatch: newStringP("tv"),
+			},
+			md:     metadata.Pairs("th", "not-match"),
+			invert: true,
+			want:   true,
+		},
```

Comment:
Please add test cases that cover the unusual corner cases mentioned above once we determine the correct behavior.

Reply:
Done. Added 3 test cases, invert and not invert for missing, and invert for empty string wanting true.